src/server_manager/web_app/app.ts

```diff
@@ -200,136 +209,218 @@ export class App {
       this.appRoot.openGetConnectedDialog(this.getS3InviteUrl(event.detail.accessKey, true));
     });
 
+    appRoot.addEventListener('ShowServerRequested', (event: PolymerEvent) => {
+      const displayServerId = event.detail.displayServerId;
+      const serverEntry = this.serverDisplayTable[displayServerId];
+      if (!serverEntry && !!this.serverBeingCreated) {
+        this.showServerCreationProgress();
+      } else if (!!serverEntry) {
+        this.showServerIfHealthy(serverEntry.server, serverEntry.displayServer);
+      } else {
+        // We should never reach this if the server display table is synced.
+        console.error(`Could not find server entry for display server ID ${displayServerId}`);
+      }
+    });
+
     onUpdateDownloaded(this.displayAppUpdateNotification.bind(this));
   }
 
   start(): void {
-    // Load manual servers from storage.
-    this.manualServerRepository.listServers().then((manualServers) => {
-      // Show any manual servers if they exist.
-      if (manualServers.length > 0) {
-        this.showManualServerIfHealthy(manualServers[0]);
-        return;
-      }
+    // Load display servers and associate them with managed and manual servers.
```

Comment:
I think we can make this function easier to understand.

Seems like what really happens is:
- fetch the last-known state of servers from localStorage
- send that to the UX
- launch a bunch of queries to update the state of each server

Let's update the comment to make that clearer.

I think we can also:
- reduce the level of indentation
- de-dupe the command a bit

e.g.:
```ts
  async start() {
    this.appRoot.serverList = await this.displayServerRepository.listServers();

    const manualServers = this.manualServerRepository.listServers();

    const accessToken = this.digitalOceanTokenManager.getStoredToken();
    const managedServers =
        accessToken ? this.enterDigitalOceanMode(accessToken) : Promise.resolve([]);

    Promise.all([manualServers, managedServers])
        .then(([a, b]) => {
          return this.syncServersToDisplay(a.concat(b));
        })
        .then(() => {
          this.maybeShowLastDisplayedServer();
        });

    this.showIntro();
  }
```